### PR TITLE
FIX game hang when units with 0 min-range attack units at 0 range.

### LINKED ISSUE #13307: HeliFly.cs activity hangs the game (zero minimum range case)
Infinite loop:

A helicopter is very likely to hover over the target when controlled by AI. In this case
* HeliFly.tick() : Try to move towards target evaluates that the helicopter is inside the maximum range of the weapon so it thinks that it may proceed to Attack activity.
* Attack.cs, InnerTick(): However, this one thinks it needs to move, because
```
			if (!Target.IsInRange(self.CenterPosition, maxRange) || Target.IsInRange(self.CenterPosition, minRange)
				|| (mobile != null && !mobile.CanInteractWithGroundLayer(self)))
```

* The first expression correctly evaluates that the helicopter is in maxRange, proceeds to the next expr.
* The second one, evaluates that target is in minimumRange.
* IsInRange is INCLUSIVE and Helicopters with minRange of 0 will think that it needs to move away out of minimum range.
* Then it goes to HeliFly.tick() to move again then comes back to attack because it is within maximum range.

There's nothing that prevents modders having 0 range for helis!
Another thing is that 0 range makes sense when the "helicopter" is an airborne aircraft carrier!

### PR DESCRIPTION
**Pull request by forcecore**
Fixes #13307

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Attack.cs [hunk 1]
```diff
@@ -102,8 +102,9 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 
 			var pos = self.CenterPosition;
 			var mobile = move as Mobile;
-			if (!Target.IsInRange(pos, maxRange) || Target.IsInRange(pos, minRange)
-				|| (mobile != null && !mobile.CanInteractWithGroundLayer(self)))
+			if (!Target.IsInRange(self.CenterPosition, maxRange)
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Attack.cs]: `self.CenterPosition` is cached as `pos` above, so why do you use `self.CenterPosition` directly?

File: OpenRA.Mods.Common/Activities/Attack.cs [hunk 2]
```diff
@@ -102,8 +102,9 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
...
+					|| (minRange.Length != 0 && Target.IsInRange(self.CenterPosition, minRange))
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Attack.cs]: Same here.

**Review by reaperrr — APPROVED** [review]
Otherwise looks good.

**Comment by forcecore** [comment]
Now using cached pos instead of self.CenterPosition.

**Comment by reaperrr** [comment]
This is a simple enough fix that should go into the playtest.

**Review by GraionDilach — APPROVED** [review]
:+1: Very well done.

File: OpenRA.Mods.Common/Activities/Attack.cs [hunk 3]
```diff
@@ -102,8 +102,9 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 
 			var pos = self.CenterPosition;
 			var mobile = move as Mobile;
-			if (!Target.IsInRange(pos, maxRange) || Target.IsInRange(pos, minRange)
-				|| (mobile != null && !mobile.CanInteractWithGroundLayer(self)))
+			if (!Target.IsInRange(pos, maxRange)
+					|| (minRange.Length != 0 && Target.IsInRange(pos, minRange))
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Attack.cs]: One level of indentation too much, I think. (Use just one tab.)

**Comment by abcdefg30** [comment]
Thanks for fixing this. The freeze is gone as advertised. 👍 / ✅

**Comment by forcecore** [comment]
Removed tabs

**Comment by reaperrr** [comment]
@forcecore This looks good now, but could you also squash the commits? For such small fixes it's better to have everything in one commit.

**Comment by forcecore** [comment]
Squashed.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8372370d92146e1ef9fc729ac68c750dd0d12e64...c9d836fc640a63777a438edfbf1f633ef1c0fd9d

**Pull request merged**